repose-aggregator/components/filters/compression/src/main/scala/org/openrepose/filters/compression/CompressionFilter.scala

```diff
@@ -0,0 +1,149 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+
+package org.openrepose.filters.compression
+
+import java.io.{EOFException, IOException}
+import java.net.URL
+import javax.inject.{Named, Inject}
+import javax.servlet._
+import javax.servlet.http.HttpServletResponse
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.servlet.http.HttpServletResponseWrapper
+import org.openrepose.commons.utils.servlet.http.ResponseMode._
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.external.pjlcompression.CompressingFilter
+import org.openrepose.filters.compression.CompressionFilter.CompressionParameters.CompressionParameters
+
+import scala.collection.JavaConverters._
+import scala.language.implicitConversions
+import scala.util.{Failure, Success, Try}
+
+@Named
+class CompressionFilter @Inject()(configurationService: ConfigurationService, compressingFilterFactory: CompressingFilterFactory)
+  extends Filter with UpdateListener[ContentCompressionConfig] with LazyLogging {
+
+  import CompressionFilter._
+  import CompressionParameters._
+
+  private var configurationFile: String = DEFAULT_CONFIG
+  private var initialized = false
+
+  // keep it around so we can initialize the filter that's doing the actual work
+  private var filterConfig: FilterConfig = _
+
+  // filter that's doing the actual work
+  private var actualFilter: CompressingFilter = _
+
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    logger.trace("Compression filter initializing...")
+    this.filterConfig = filterConfig
+    configurationFile = new FilterConfigHelper(filterConfig).getFilterConfig(DEFAULT_CONFIG)
+
+    logger.info(s"Initializing Compression filter using config $configurationFile")
+    val xsdUrl: URL = getClass.getResource(SCHEMA_FILE_NAME)
+    configurationService.subscribeTo(filterConfig.getFilterName, configurationFile, xsdUrl, this, classOf[ContentCompressionConfig])
+
+    logger.trace("Compression filter initialized.")
+  }
+
+  override def doFilter(request: ServletRequest, response: ServletResponse, filterChain: FilterChain): Unit = {
+    val responseWrapper: HttpServletResponseWrapper = new HttpServletResponseWrapper(
```

Comment:
I don't see why you are wrapping it? The only method call i see is `setStatus` which is a standard part of the `HttpsServletResponse` contract